app/services/classification/matches_generator.rb

```diff
@@ -65,13 +66,13 @@ def is_parameter?(element)
       host.host_params.include?(element)
     end
 
-    def hostgroup_matches
+    def inherited_matches(element)
       matches = []
-      if host.hostgroup
-        path = host.hostgroup.to_label
+      if is_method?(element) && host.send(element)
```

Comment:
Do we really need to check `is_method?` on host for these methods?

Any scenario where those methods won't be there on host instance? 
In case it is required then would be good to filter out set of inheritable matchers at first place using `is_method?` & `add_inherited_matches?`

Reply:
Has taxonomies are now mandatory, yes I think you are right, this check is not mandatory anymore.